openedx/openedx-platform
# feat: make video feature api to be course agnostic

## Description

This PR removes the course id requirement for the video feature api. The video feature api is needed in libraries, which don't have course ids, to determine if social sharing is allowed. This change impacts Developers.

## Supporting information

JIRA Ticket: [AU-1188](https://2u-internal.atlassian.net/browse/AU-1188)

> The video xblock editor must respect the course level video sharing state. Follow the linked Figma design.
> 
> AC
> 1. Reads the current video share state from the course (possibly course video features view)
> 2. Move sharing checkbox into own section (see figma)
> 3. Don’t show section if feature is not enabled for course
> 4. Checkbox should reflect the current state for the field for the xblock but should be greyed out / disabled with message if overridden by course setting (see figma)

## Testing instructions

1. Checkout [frontend-lib-content-componets PR 324](https://github.com/openedx/frontend-lib-content-components/pull/324)
2. In Studio Django admin enable `video_config.public_video_share`.
3. Create a video xblock in a library.
4. Check that the social sharing section is visible.
5. In Studio Django admin disable `video_config.public_video_share`.
6. Navigate back to video xblock and check that the social sharing section is not visible.

## Deadline

None

## Other information

This PR is connected to [frontend-lib-content-componets PR 324](https://github.com/openedx/frontend-lib-content-components/pull/324).
